File: examples/payments/ledger/src/main/java/org/portolan/payments/ledger/infrastructure/oms/event/OrderCancelled.java
package org.portolan.payments.ledger.infrastructure.oms.event;

/**
 * Another service's event, in the shape this one needs to read it.
 *
 * The manifest says which aggregate it belongs to; without that line the policy
 * that listens for it names a type and resolves to nothing.
 */
public record OrderCancelled(String orderId, String reason) {

    public static final String NAME = "oms.OrderCancelled";
}

File: plugins/extract-rust/testdata/oms/src/domain/order/event/order_cancelled.rs
use super::Event;

/// The order will not be fulfilled.
pub struct OrderCancelled {
    pub order_id: String,
    pub reason: String,
}

impl Event for OrderCancelled {
    fn name(&self) -> &'static str {
        "oms.OrderCancelled"
    }
}
